x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/BatchedDocumentsIterator.java

```diff
@@ -89,6 +90,8 @@ public boolean hasNext() {
     private SearchResponse initScroll() {
         LOGGER.trace("ES API CALL: search index {}", index);
 
+        Transports.assertNotTransportThread("BatchedDocumentsIterator makes blocking calls");
```

Comment:
preferably put this into the `next()` method instead so it will also cover the other blocking calls in this class.
Could you also write this as `assert Transports.assertNotTransportThread(...)`, this will save from extra CPU in non-debug mode.

Reply:
I wonder if we need this at all. The blocking call to the client executes it anyway. The issue was that there was no testing. I think this entire transport action can use a ml threadpool instead

Reply:
True, it's just a more helpful error message

Reply:
Please put it into an assert if you keep it. I’d remove it.